sourcegraph/cody
# Edit: Allow editing the inline diff

## Description

closes https://github.com/sourcegraph/cody/issues/4671


https://github.com/sourcegraph/cody/assets/9516420/7d468713-c928-461b-b0f0-192a91560d44


Problem:
- We auto-accept the task whenever there is _any_ change within the `selectionRange`.
   - This was primarily useful for tracking if the user immediately does "Undo" on a change, so we update the UI and clear out the diff.
- With the new inline diff view, doing this means that it's difficult for users' to modify Cody's response. As any change they make immediately removes all deleted lines and hide decorations.

Solution:
- Allow making edits within the selection range. We can ensure decorations do not expand so it's still clear exactly what edits Cody made.
- Track specifically for the edit being restored back to its original state. This will work to ensure we still capture when a user does an "Undo"

## Test plan

1. Make Edits (Generate, Edit, Test, Fix, Document)
2. Make changes to the diff
3. Check everything looks ok
4. Try to undo _some_ of the changes you made
5. Check the entire edit is not removed, as we're only undoing additional changes
6. Undo all the changes made after the edit
7. Check the entire edit is removed (e.g. codelens, injected whitespace, etc)
